src/test/java/roomescape/application/ReservationServiceTest.java

```diff
@@ -0,0 +1,143 @@
+                ReservationDateFixture.예약날짜_25_4_22.getDate(),
+                reservationTime.getId(),
+                theme.getId()
+        ));
+
+        ReservationCreateRequest request = new ReservationCreateRequest(
+                ReserverNameFixture.한스.getName(),
+                ReservationDateFixture.예약날짜_25_4_22.getDate(),
+                reservationTime.getId(),
+                theme.getId()
+        );
+
+        assertThatThrownBy(() -> reservationService.createReservation(request))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void 같은일시_다른테마_예약은_생성할_수_있다() {
+        ReservationTime reservationTime = reservationTimeDbFixture.예약시간_10시();
+        Theme horror = themeDbFixture.공포();
+        Theme mystery = themeDbFixture.커스텀_테마("미스테리");
+
+        reservationService.createReservation(new ReservationCreateRequest(
+                ReserverNameFixture.한스.getName(),
+                ReservationDateFixture.예약날짜_25_4_22.getDate(),
+                reservationTime.getId(),
+                horror.getId()
+        ));
+
+        ReservationCreateRequest request = new ReservationCreateRequest(
+                ReserverNameFixture.한스.getName(),
+                ReservationDateFixture.예약날짜_25_4_22.getDate(),
+                reservationTime.getId(),
+                mystery.getId()
+        );
+
+        assertThatCode(() -> reservationService.createReservation(request))
+                .doesNotThrowAnyException();
+    }
+
+    @Test
+    void 예약을_모두_조회한다() {
+        ReservationTime reservationTime = reservationTimeDbFixture.예약시간_10시();
+        Theme theme = themeDbFixture.공포();
+
+        reservationService.createReservation(new ReservationCreateRequest(
+                ReserverNameFixture.한스.getName(),
+                ReservationDateFixture.예약날짜_25_4_22.getDate(),
+                reservationTime.getId(),
+                theme.getId()
+        ));
+        List<ReservationResponse> responses = reservationService.getReservations();
+        ReservationResponse response = responses.get(0);
+
+        assertThat(response.id()).isEqualTo(1L);
+        assertThat(response.name()).isEqualTo(ReserverNameFixture.한스.getName());
+        assertThat(response.date()).isEqualTo(ReservationDateFixture.예약날짜_25_4_22.getDate());
+        assertThat(response.time()).isEqualTo(
+                new ReservationTimeResponse(reservationTime.getId(), reservationTime.getStartAt().toString()));
+    }
+
+    @Test
+    void 예약을_삭제한다() {
+        ReservationTime reservationTime = reservationTimeDbFixture.예약시간_10시();
+        Theme theme = themeDbFixture.공포();
+
+        reservationService.createReservation(new ReservationCreateRequest(
+                ReserverNameFixture.한스.getName(),
+                ReservationDateFixture.예약날짜_25_4_22.getDate(),
+                reservationTime.getId(),
+                theme.getId()
+        ));
+        reservationService.deleteReservationById(1L);
```

Comment:
어떤 id 값이 나올지 추측해서 코드를 테스트 코드를 짜는 것 보다는,  
생성한 결과에서 id를 추출해서 삭제 요청에 넣으면 되지 않을까요? 

추가적으로 test 코드또한 유지보수의 대상이므로 가능하면  테스트 코드 작성 시 given, when, then  구조를 한눈에 파악할 수 있도록 작성하는 연습을 해보시면 좋을 것 같아요. 

예를들어 지금의 테스트 케이스는 어떤 메서드를 테스트하려는 것인지 한눈에 들어오지가 않는 것 같아요. 👀

```suggestion
        var createdReservation = reservationService.createReservation(new ReservationCreateRequest(
                ReserverNameFixture.한스.getName(),
                ReservationDateFixture.예약날짜_25_4_22.getDate(),
                reservationTime.getId(),
                theme.getId()
        ));
        reservationService.deleteReservationById(createdReservation.id());
```

Reply:
맞아요! 로키의 의견에 동의합니다.
Fixture를 여러 개 도입해서 사용하는 것은 처음 시도해본 것이어서 정신없이 테스트 코드를 짰던 것 같아요.🥲
생성한 객체의 id를 추출해서 삭제 요청에 넣도록 수정해두었습니다.
given, when, then 구조를 조금 더 명확히하는 리팩터링은 아래 피드백에 함께 반영했습니다!

[refactor: 생성된 결과에서 id를 추출하여 삭제하도록 수정](https://github.com/ljhee92/spring-roomescape-member/commit/4c5320dac4482f9bdd1b73e43d799d2dc985577c)